### ISSUE TITLE
General Issue - Unstable Orbits

### ISSUE THREAD

**Issue by shadowmage45** [shadowmage45/SSTULabs#309]
While doing testing I noticed -very- unstable orbits for a few craft while in low Mun orbit;  started at 50x50 circular orbit and within a few minutes orbit had... degraded... to 57x35.  Seems much worse on a lander that has stock landing gear on it, though it also occurs with a simple SC-B CSM stack.

Had noticed similar problems in the past when KJR is in use (both in 0.90 and in 1.0.x).  It seemed that more massive vessels in orbit in lower-gravity situations exacerbated the problem.

Need to verify that this is or is not caused by specific SSTU parts;  will need to remove KJR to make sure it is not the culprit (again), and test a few stock-only craft in similar orbits, comparing vs SSTU derived craft.

If it is SSTU parts I have little/no idea what could be causing it or any potential solutions; nothing in SSTU adds forces to a craft in space.

**Comment by shadowmage45** [comment]
Seems to be a known stock issue; though the amount of instability I'm seeing is a bit worse than others are reporting.

Stock bug report can be found at: http://bugs.kerbalspaceprogram.com/issues/9619

Will do a slight bit of investigation tonight, but likely I'll merely confirm that there is a problem with stock (that is exaggerated by mods and/or KJR).

As it stands though docking is nearly impossible as the craft do not obey physics and are undergoing constant random accelerations and torques;  I have to basically run the game in 5x timewarp and drop to normal time to do course corrections and the final docking maneuver.

**Comment by shadowmage45** [comment]
A bit of investigation reveals that this, at least in my case, is mostly caused by KJR.  A vessel that was very much instable (the mun lander/station from above), is reasonably stable once KJR has been removed.  No more phantom torques, no more phantom accelerations, and is capable of docking without 'chasing' the impossibly-accelerating craft.

If I remove KJR then the orbits are much more stable;  there is still slight drift, but it is in the order of meters per minute rather than kilometers per minute, and is mostly cyclical (stays within a range of the original value).

If this is indeed the case...(KJR causing phantom forces), and it appears to be so, I'm likely going to have to give up on playing KSP entirely, as it cannot be played without KJR (noodle rockets make me rage), and cannot actually be played with it either (unstable orbits; cannot play a physics based game if physics is screwey).

Either way, not an issue with SSTU or its parts, but is an issue with 1.) the stock code/physx/unity, and 2.) KJR added joints (likely a constraints or harmonics issue with multi-jointed parts)

**Comment by blowfishpro** [comment]
Had a chat with ferram4 about this and it sounds like it might be worth reporting although there's a good chance it can't be fixed.  Best would be a stock craft that easily exhibits the issue and maybe a save file with the craft already in orbit (to avoid Hyperedit issues).  If you don't get to this I can probably try in the next few days.

**Comment by shadowmage45** [comment]
Thanks, and noted.  I have a long weekend coming up, so should have time to put something together in the next couple of days;  more info is always good though.

Did a ton of debugging on this back in the 0.90 days with larger interplanetary craft and stations (before I even started modding), so I've got a good idea on a few stock setups that will exhibit the problem (or did at least).  I've always taken it as a 'can't be solved' problem and a side-effect of the added joints, but it likely is worth at least opening an ticket for, so it can be a 'known issue' if nothing else.

I also wouldn't be surprised if some of the SSTU parts are aggravating the problem due to the strange module combinations that some of them have; but last I played with this problem it could be easily duplicated with stock parts on some reasonably built craft (and some of the stock craft).

Edit:  Indeed, can easily be seen on the upper CSM portion of the stock Kerbal-X craft:
Started in a 50x50 orbit, and am losing about 500m from my periapsis every minute of flight (time shown in UI is pretty much the time spent in orbit); additionally there is a constant torque on the craft and it will constantly increase its angular velocity if it is not countered.

If I have either MJ or stock SAS hold an orientation the rate of the orbit adjustment increases substantially, it is worse with MJ than stock systems.

![screenshot22](https://cloud.githubusercontent.com/assets/1503399/15201207/b192d8e2-17ab-11e6-9023-b10e92c5b44e.png)
![screenshot23](https://cloud.githubusercontent.com/assets/1503399/15201208/b19429ae-17ab-11e6-825a-c1fe177c3b43.png)
![screenshot24](https://cloud.githubusercontent.com/assets/1503399/15201209/b1965b52-17ab-11e6-85d7-e7380d9ae838.png)

etc....

Will work on putting this all together with some logs and submitting it for the bug report.

Some of this may also be a stock issue as there have been a few reports of it on the forums in stock installs; though it is much less severe from the testing that I have done.
